src/ui/components/EntryIcon/EntryIcon.tsx

```diff
@@ -122,24 +122,21 @@ const entityTypeIcons: Record<string, string> = {
 
 const getScopeTypeIcon = (scope: string) => {
     switch (scope) {
-        case EntryScope.Folder:
-            return 'folder';
         case EntryScope.Widget:
         case 'chart':
             return 'chart-wizard';
-        case EntryScope.Dataset:
-            return 'dataset';
         case EntryScope.Dash:
         case 'dashboard':
             return 'dashboard';
         case 'monitoring':
             return 'editor';
-        case EntryScope.Report:
-            return 'report';
         case 'broken':
             return 'broken';
         default:
-            return null;
+            // For all new scopes icon type should match scope:
+            // EntryScope.Dataset -> 'dataset'
+            // EntryScope.Folder -> 'folder'
+            return scope;
```

Comment:
What about 'unknown' types? Before that `getEntityIconType` was aiming for `null` - now there is no whitelist at all.

Reply:
Normally getEntityIconType should not return null

Reply:
https://github.com/datalens-tech/datalens-ui/blob/a4eec11f9ef55e0647da5fdca0797f1a7ebbe710/src/ui/components/EntryIcon/EntryIcon.tsx#L184 - but it does and component that renders this Icon is done to work with `null` value. Current method is working as whitelist filter so that we wouldn't try to get icon for uknown Entry type.